### DISCUSSION TITLE
`brew cask install` on Apple Sillicon

### DISCUSSION THREAD

**Discussion by vojkny** [Homebrew/discussions#2078] [Casks]
I wonder how brew distinguishes correct platform for cask installation.

1. I have migrated my Apple installation from Intel platform to Apple Sillicon platform and then running `brew upgrade` had no effect - I would expect it to reinstall correct versions for the new platform.
2. I then tried to remove IntelliJ Idea and reinstall it with brew, but it still installed the `Intel` version.

Is there something that needs to be manually changed?

**Comment by SMillerDev** [comment]
> I have migrated my Apple installation from Intel platform to Apple Sillicon platform

How did you migrate this? If you used a Timemachine backup then you didn't migrate anything, you simply moved Intel software to your new ARM machine.
If you want to set up Homebrew for ARM, run the installer on `brew.sh` and reinstall all packages in ARM mode (so no Rosetta).

**Reply by vojkny** [reply]
Yes, I used timemachine, so what would be the correct way to reinstall everything?

**Reply by SMillerDev** [reply]
`brew bundle` is the best tool for that.

**Reply by vojkny** [reply]
I do use brew bundle, I have a huge list for it. But it still does not do anything after the timemachine migration to APple sillicon.

**Reply by vojkny** [reply]
Ideally I would expect the brew to recognize a different platform and reinstall on upgrade.

**Reply by unitof** [reply]
That probably won't happen. Instead, just run `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"` on your new M1 machine, then once `brew` is installed run `brew bundle install`.

**Reply by SMillerDev** [reply]
> Ideally I would expect the brew to recognize a different platform and reinstall on upgrade.

Since not all software is M1 compatible we can't do that.

As for the `brew bundle` make sure that you're not running your terminal and/or brew in Rosetta mode after installing again. Otherwise it'll install Intel software again.

**Reply by carlocab** [reply]
Don't run the install script under Rosetta too.
